src/rviz/default_plugin/covariance_visual.cpp

```diff
@@ -300,6 +300,13 @@ void CovarianceVisual::setCovariance( const geometry_msgs::PoseWithCovariance& p
 
   // store orientation in Ogre structure
   Ogre::Quaternion ori(pose.pose.orientation.w, pose.pose.orientation.x, pose.pose.orientation.y, pose.pose.orientation.z);
+  // in the case that a null quaternion is in the message (e.g. if the field is uninitialized),
+  // set it to identity to prevent crashes
+  if (0.0 == ori.x && 0.0 == ori.y && 0.0 == ori.z && 0.0 == ori.w)
+  {
+    ori.w = 1.0;
+  }
+
   // Set the orientation of the fixed node. Since this node is attached to the root node, it's orientation will be the
   // inverse of pose's orientation.
   fixed_orientation_node_->setOrientation(ori.Inverse());
```

Comment:
In contrast to #1179, this PR doesn't handle normalization of the quaternion. As `ogre` and `rviz/ros` might follow different normalization strategies, this might yield inconsistent results, because the quaternion is once normalized internally by ROS (see https://github.com/ros-visualization/rviz/pull/1179#issuecomment-354971379) before being passed to `SceneNode::setOrientation` and once (here) not.